packages/compartment-mapper/src/import-hook.js

```diff
@@ -720,6 +712,7 @@ export function makeImportNowHookMaker(
           packageLocation,
           packageSources,
           readPowers,
+          archiveOnly,
```

Comment:
I did not need to reveal a new option to `makeBundle` in order to indicate whether a `sourceURL` comment should be generated at the time a module is parsed off disk, because we already had `archiveOnly`, which is not orthogonal. I did get caught surprised that we _do_ need to inject a `sourceURL` at parse time in order to preserve stack traces for `importLocation`, since that doesn’t go through the archiver or bundler.